runtime/compiler/runtime/CRRuntime.cpp

```diff
@@ -148,6 +157,177 @@ TR::CRRuntime::waitOnCRRuntimeMonitor()
    _crRuntimeMonitor->wait();
    }
 
+void
+TR::CRRuntime::pushMemoizedCompilation(TR_MemoizedCompilations& list, J9Method *method)
+   {
+   auto newEntry = new (_compInfo->persistentMemory()) TR_MemoizedComp(method);
+   if (newEntry)
+      list.add(newEntry);
+   }
+
+J9Method *
+TR::CRRuntime::popMemoizedCompilation(TR_MemoizedCompilations& list)
+   {
+   J9Method *method = NULL;
+   auto memComp = list.pop();
+   if (memComp)
+      {
+      method = memComp->getMethod();
+      jitPersistentFree(memComp);
+      }
+   return method;
+   }
+
+void
+TR::CRRuntime::removeMemoizedCompilation(TR_MemoizedCompilations& list, J9Method *method)
+   {
+   auto first = list.getFirst();
+   if (!first)
+      {
+      return;
+      }
+   else if (first->getMethod() == method)
+      {
+      popMemoizedCompilation(list);
+      }
+   else
+      {
+      auto prev = first;
+      auto curr = prev->getNext();
+
+      while (curr)
+         {
+         if (curr->getMethod() == method)
+            {
+            list.removeAfter(prev, curr);
+            jitPersistentFree(curr);
+            break;
```

Comment:
This assumes that the list cannot have duplicates. I haven't checked whether we enforce that.

Reply:
There technically can't be any duplicates, but nothing enforces it. However, as you mentioned, scanning the list using the class addresses the duplicates.